File: 0020-Valid-Parenthese/0020.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for char in s:
            if char == '(' or char == '{' or char == '[':
                stack.append(char)
            elif char == ')' and len(stack) > 0 and stack[len(stack) - 1] == '(':
                stack.pop()
            elif char == '}' and len(stack) > 0 and stack[len(stack) - 1] == '{':
                stack.pop()
            elif char == ']' and len(stack) > 0 and stack[len(stack) - 1] == '[':
                stack.pop()
            else: 
                return False
            
        return len(stack) == 0
            

File: 0020-Valid-Parenthese/0020.java
import java.util.Stack;

class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        char[] charArray = s.toCharArray();
        for (char c : charArray) {
            if (c == '(' || c == '{' || c == '[') {
                stack.push(c);
            } else if (!stack.empty() && c == ')' && stack.peek() == '(') {
                stack.pop();
            } else if (!stack.empty() && c == '}' && stack.peek() == '{') {
                stack.pop();
            } else if (!stack.empty() && c == ']' && stack.peek() == '[') {
                stack.pop();
            } else {
                return false;
            }
        }
        return stack.empty();
    }
}